include/tbb/machine/linux_common.h

```diff
@@ -26,8 +26,11 @@
 #define __TBB_Yield()  sched_yield()
 
 #include <unistd.h>
+
+#ifndef __HAIKU__
 /* Futex definitions */
 #include <sys/syscall.h>
+#endif
 
 #if defined(SYS_futex)
```

Comment:
It looks like that `SYS_futex`  can be defined on Haiku. If not, move `#endif` on line 33 to the end of the file.